### PR TITLE
🐛 Fixed Expired JWT Token By Refreshing it

### PR DESCRIPTION
**Pull request by Aibono1225**
Relates to #1340 

Changes:
- Enabled refresh tokens for all APIs requiring authorization
- Added a prompt to "log in again" if the refresh token expires

image
**Figure: Settings in Auth0 dashboard**

### CODE REVIEW

File: src/components/layout/layout.js [hunk 1]
```diff
@@ -38,10 +38,11 @@ const Layout = ({
         authorizationParams={{
           redirect_uri: process.env.AUTH0_REDIRECT_URI,
         }}
-        // redirectUri={process.env.AUTH0_REDIRECT_URI}
         onRedirectCallback={onRedirectCallback}
         useRefreshTokens={true}
         cacheLocation="localstorage"
+        scope="openid profile email offline_access"
```
Comment (JackDevAU) [inline src/components/layout/layout.js:44]: lets put these in the .env

File: src/services/authService.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const fetchIdToken = async () => {
+    try {
+      const claims = await getIdTokenClaims();
+      const expiryTime = claims.exp * 1000;
+      const currentTime = new Date().getTime();
+
+      if (expiryTime - currentTime < 60000) {
+        await getAccessTokenSilently({
+          audience: process.env.AUTH0_DOMAIN,
+          scope: 'openid profile email offline_access',
```
Comment (JackDevAU) [inline src/services/authService.js:16]: ditto of the above - should these be in a `.env` file?

File: src/components/layout/layout.js (hunk 1 above)
Comment (Aibono1225) [reply src/components/layout/layout.js]: sounds good! Done

**Review by JackDevAU — APPROVED** [review]

**Review by bradystroud — APPROVED** [review]
LGTM 
I assume the `AUTH0_SCOPE` has been added to the github secrets so it works on production

**Comment by Aibono1225** [comment]
> LGTM I assume the `AUTH0_SCOPE` has been added to the github secrets so it works on production

Yes they are now in github variables 🙂

**Pull request merged**